src/interfaces/python/swig_typemaps.i

```diff
@@ -1292,5 +1292,30 @@ TYPEMAP_SPARSEFEATURES_OUT(PyObject,      NPY_OBJECT)
     PyErr_SetString(PyExc_SystemError, $1.what());
     SWIG_fail;
 }
+%rename(_kernel) kernel;
+
+%pythoncode %{
+def _internal_factory_wrapper(object_name, new_name, docstring=None):
+    """
+    A wrapper that returns a generic factory that
+    accepts kwargs and passes them to shogun.object_name
+    via .put
+    """
+    _obj = getattr(_shogun, object_name)
```

Comment:
maybe add a comment that says "Instantiate object via shogun factory, called by string"...this might be something where other stumble over ... or maybe it is self explaining. What do you think?